contracts/anchor_basset_reward/src/global.rs

```diff
@@ -27,15 +27,16 @@ pub fn handle_swap<S: Storage, A: Api, Q: Querier>(
     let reward_denom = config.reward_denom;
 
     for coin in balance {
-        if coin.denom == reward_denom {
+        if coin.denom == reward_denom.clone() {
             continue;
         }
-
-        msgs.push(create_swap_msg(
-            contr_addr.clone(),
-            coin,
-            reward_denom.to_string(),
-        ));
+        if query_exchange_rates(deps, reward_denom.clone(), vec![coin.denom.clone()]).is_ok() {
```

Comment:
I think we can optimize this by querying all denoms exchange_rates first. and if something failed, we can try each one.